archive/demos/shiny-drum-machine.js

```diff
@@ -424,118 +203,118 @@ function init() {
-    document.getElementById('snare_thumb').addEventListener('dblclick', handleSliderDoubleClick, true);
-    document.getElementById('kick_thumb').addEventListener('dblclick', handleSliderDoubleClick, true);
-    document.getElementById('swing_thumb').addEventListener('dblclick', handleSliderDoubleClick, true);
+    ui.notes = new Notes();
+    ui.notes.onClick = (instrumentName, rhythm) => handleNoteClick(instrumentName, rhythm);
 
-    // tool buttons
-    document.getElementById('play').addEventListener('click', handlePlay, true);
-    document.getElementById('stop').addEventListener('click', handleStop, true);
-    document.getElementById('save').addEventListener('click', handleSave, true);
-    document.getElementById('save_ok').addEventListener('click', handleSaveOk, true);
-    document.getElementById('load').addEventListener('click', handleLoad, true);
-    document.getElementById('load_ok').addEventListener('click', handleLoadOk, true);
-    document.getElementById('load_cancel').addEventListener('click', handleLoadCancel, true);
-    document.getElementById('reset').addEventListener('click', handleReset, true);
-    document.getElementById('demo1').addEventListener('click', handleDemoMouseDown, true);
-    document.getElementById('demo2').addEventListener('click', handleDemoMouseDown, true);
-    document.getElementById('demo3').addEventListener('click', handleDemoMouseDown, true);
-    document.getElementById('demo4').addEventListener('click', handleDemoMouseDown, true);
-    document.getElementById('demo5').addEventListener('click', handleDemoMouseDown, true);
-
-    document.getElementById('tempoinc').addEventListener('click', tempoIncrease, true);
-    document.getElementById('tempodec').addEventListener('click', tempoDecrease, true);
-}
+    ui.kitPicker = new KitPicker();
+    ui.kitPicker.addOptions(KITS.map(kit => kit.prettyName));
+    ui.kitPicker.onSelect = (i) => handleKitChange(i);
+
+    ui.effectPicker = new EffectPicker();
+    ui.effectPicker.addOptions(impulseResponses.map(e => e.name));
+    ui.effectPicker.onSelect = (i) => handleEffectChange(i);
+
+    ui.effectSlider = new EffectSlider();
+    ui.effectSlider.onchange = (value) => {
+        // Change the volume of the convolution effect.
+        theBeat.effectMix = value;
+        setEffectLevel(theBeat);
+    };
+
+    ui.swingSlider = new SwingSlider();
+    ui.swingSlider.onchange = (value) => {
+        theBeat.swingFactor = value;
+    };
 
-function initButtons() {        
-    var elButton;
+    ui.pitchSliders = new PitchSliders();
+    ui.pitchSliders.onPitchChange = (instrumentName, pitch) => setPitch(instrumentName, pitch);
 
-    for (i = 0; i < loopLength; ++i) {
-        for (j = 0; j < kNumInstruments; j++) {
-                elButton = document.getElementById(instruments[j] + '_' + i);
-                elButton.addEventListener("click", handleButtonMouseDown, true);
+    // tool buttons
```

Comment:
I think we prefer to have comments explains "why", not "what". What does this comment mean?

Reply:
This comment was leftover from the old code - removed.